spec/integrity/middleware_order_spec.rb

```diff
@@ -1,49 +1,47 @@
 # frozen_string_literal: true
 
 RSpec.describe "Middleware order" do
-  it "order of middleware in the application is correct" do
-    middlewares = Rails.configuration.middleware.map { |middleware| "#{middleware.inspect}" }
-    expect(middlewares).to eq(
-      %w[
-        BlockRequestsMiddleware
-        TestMultisiteMiddleware
-        ActionDispatch::RemoteIp
-        Middleware::RequestTracker
-        MessageBus::Rack::Middleware
-        ActionDispatch::HostAuthorization
-        Rack::Sendfile
-        ActionDispatch::Static
-        ActionDispatch::Executor
-        Rack::MethodOverride
-        Middleware::EnforceHostname
-        ActionDispatch::RequestId
-        SilenceLogger
-        ActionDispatch::ShowExceptions
-        ActionDispatch::DebugExceptions
-        ActionDispatch::Callbacks
-        ActionDispatch::Cookies
-        ActionDispatch::Session::DiscourseCookieStore
-        Discourse::Cors
-        ActionDispatch::Flash
-        RspecErrorTracker
-        Middleware::CspScriptNonceInjector
-        Middleware::AnonymousCache
-        ContentSecurityPolicy::Middleware
-        ActionDispatch::PermissionsPolicy::Middleware
-        Rack::Head
-        Rack::ConditionalGet
-        Rack::TempfileReaper
-        Middleware::OmniauthBypassMiddleware
-      ],
-    )
+  let(:expected_middlewares) do
+    [
+      BlockRequestsMiddleware,
+      TestMultisiteMiddleware,
+      ActionDispatch::RemoteIp,
+      Middleware::RequestTracker,
+      MessageBus::Rack::Middleware,
+      Rack::Sendfile,
+      ActionDispatch::Static,
+      ActionDispatch::Executor,
+      Rack::MethodOverride,
+      Middleware::EnforceHostname,
+      ActionDispatch::RequestId,
+      SilenceLogger,
+      ActionDispatch::ShowExceptions,
+      ActionDispatch::DebugExceptions,
+      ActionDispatch::Callbacks,
+      ActionDispatch::Cookies,
+      ActionDispatch::Session::DiscourseCookieStore,
+      Discourse::Cors,
+      ActionDispatch::Flash,
+      RspecErrorTracker,
+      Middleware::CspScriptNonceInjector,
+      Middleware::AnonymousCache,
+      ContentSecurityPolicy::Middleware,
+      ActionDispatch::PermissionsPolicy::Middleware,
+      Rack::Head,
+      Rack::ConditionalGet,
+      Rack::TempfileReaper,
+      Middleware::OmniauthBypassMiddleware,
+    ]
+  end
+  let(:actual_middlewares) { Rails.configuration.middleware.middlewares }
+  let(:remote_ip_index) { actual_middlewares.index(ActionDispatch::RemoteIp) }
+  let(:request_tracker_index) { actual_middlewares.index(Middleware::RequestTracker) }
+
+  it "has the correct order of middlewares" do
+    expect(actual_middlewares).to eq(expected_middlewares)
```

Comment:
I extracted the list of middlewares to a `let` because it was way easier to debug and see which one was missing (the diff wasn’t clear in the output), I just had to do `expected_middlewares - actual_middlewares`.

pinging @martin-brennan since you did that code yesterday :grin: